craft_application/services/remotebuild.py

```diff
@@ -276,17 +281,19 @@ def _new_repository(
 
     def _get_repository(self) -> launchpad.models.GitRepository:
         """Get an existing repository on Launchpad."""
-        return launchpad.models.GitRepository.get(
-            self.lp, name=self._name, owner=self.lp.username
-        )
+        return self.lp.get_repository(name=self._name, owner=self.lp.username)
 
-    def _new_recipe(
+    def _ensure_recipe(
         self,
         name: str,
         repository: launchpad.models.GitRepository,
         **kwargs: Any,  # noqa: ANN401
     ) -> launchpad.models.Recipe:
         """Create a new recipe."""
+        with contextlib.suppress(launchpadlib.errors.HTTPError):
```

Comment:
What is the reasoning or implication for suppressing HTTP errors here?

Reply:
There are 3 different ways this can fail in expected ways, all of them HTTPErrors:

1. Recipe doesn't exist (or you don't have access to see it)
2. You don't have access to delete the recipe
3. There is an ongoing build with the recipe

In case 1, we don't care and are about to make the new recipe anyway.
In the other cases, the errors from `new()` end up being more useful.


And because this was sufficiently complex, I made a comment explaining that.

Reply:
:point_right: :point_left: Are you going to push the commit with the comment?